test/gitea_test.go

```diff
@@ -568,6 +568,38 @@ func TestGiteaWithCLIGeneratePipeline(t *testing.T) {
 	}
 }
 
+func TestGiteaCancelRun(t *testing.T) {
+	topts := &tgitea.TestOpts{
+		TargetEvent: options.PullRequestEvent,
+		YAMLFiles: map[string]string{
+			".tekton/pr.yaml": "testdata/pipelinerun_long_running.yaml",
+		},
+		NoCleanup:    true,
+		ExpectEvents: false,
+	}
+	defer tgitea.TestPR(t, topts)()
+
+	// let pipelineRun start and then cancel it
+	time.Sleep(time.Second * 5)
+	tgitea.PostCommentOnPullRequest(t, topts, "/cancel")
+
+	waitOpts := twait.Opts{
+		RepoName:        topts.TargetNS,
+		Namespace:       topts.TargetNS,
+		MinNumberStatus: 1,
+		PollTimeout:     twait.DefaultTimeout,
+		TargetSHA:       topts.PullRequest.Head.Sha,
+	}
+	err := twait.UntilRepositoryUpdated(context.Background(), topts.Clients.Clients, waitOpts)
+	assert.Error(t, err, "pipelinerun has failed")
+
+	prs, err := topts.Clients.Clients.Tekton.TektonV1beta1().PipelineRuns(topts.TargetNS).List(context.Background(), metav1.ListOptions{})
```

Comment:
@chmouel any thought on this test?
deferred func is exectution before this line executes

Reply:
may be i can use sync group 🤔